modules/platform.py

```diff
@@ -431,21 +463,34 @@ def get_all_audits(self, **kwargs):
         :param event_object:
         :param event_action:
         :param limit:
+        :param ordering:
         """
-        resp = self._request(
-            method="get",
-            url=f"{self.url}/events/audits",
-            params=self._generate_audit_params(kwargs),
-        )
+        try:
+            limit = kwargs.pop("limit")
+        except KeyError:
+            limit = 500
+
+        if limit >= 1000:
+            limit = 1000
+        elif limit <= 0:
+            limit = 1
+
+        kwargs["limit"] = limit
+        resp = self.get_audits(**kwargs)
         yield resp
-        while resp.json()["remaining_count"] > 0:
-            kwargs["checkpoint"] = resp.json()["next_checkpoint"]
-            resp = self._request(
-                method="get",
-                url=f"{self.url}/events/audits",
-                params=self._generate_audit_params(kwargs),
-            )
-            yield resp
+
+        if self.threads == 1:
```

Comment:
Minor suggestion, this code block to handle threaded vs non-threaded may work better as a shared function/method